src/core/tests/type_prop/read_value.cpp

```diff
@@ -15,4 +15,84 @@ TEST(type_prop, read_value_deduce) {
+    auto variable_info = op::util::VariableInfo{PartialShape{1, 2, 64, 64}, element::f32, "variable_id"};
+    auto variable = std::make_shared<op::util::Variable>(variable_info);
+    std::shared_ptr<ov::op::v6::ReadValue> read_value;
+    EXPECT_ANY_THROW(read_value = std::make_shared<ov::op::v6::ReadValue>(input, variable));
+}
+
+TEST(type_prop, read_value_v6_dyn_shape_type_in_variable) {
+    auto input = make_shared<ov::op::v0::Parameter>(element::f32, Shape{1, 2, 64, 64});
+
+    auto variable_info = op::util::VariableInfo{PartialShape{Dimension::dynamic(), 2, Dimension::dynamic(), 64},
+                                                element::dynamic,
+                                                "variable_id"};
+    auto variable = std::make_shared<op::util::Variable>(variable_info);
+
+    std::shared_ptr<ov::op::v6::ReadValue> read_value;
+    EXPECT_NO_THROW(read_value = std::make_shared<ov::op::v6::ReadValue>(input, variable));
+
+    ASSERT_EQ(read_value->get_element_type(), element::dynamic);
+    ASSERT_EQ(read_value->get_output_partial_shape(0),
+              (PartialShape{Dimension::dynamic(), 2, Dimension::dynamic(), 64}));
+    ASSERT_EQ(read_value->get_variable_id(), "variable_id");
+}
+
+TEST(type_prop, read_value_v6_init_shape_is_in_range) {
+    auto input = make_shared<ov::op::v0::Parameter>(element::f32, PartialShape{1, 2, 64, 64});
+
+    auto variable_info =
+        op::util::VariableInfo{PartialShape{{1, 10}, {2, 5}, {64, 64}, 64}, element::f32, "variable_id"};
+    auto variable = std::make_shared<op::util::Variable>(variable_info);
+
+    std::shared_ptr<ov::op::v6::ReadValue> read_value;
+    EXPECT_NO_THROW(read_value = std::make_shared<ov::op::v6::ReadValue>(input, variable));
+
+    ASSERT_EQ(read_value->get_element_type(), element::f32);
+    ASSERT_EQ(read_value->get_output_partial_shape(0), (PartialShape{{1, 10}, {2, 5}, {64, 64}, 64}));
+    ASSERT_EQ(read_value->get_variable_id(), "variable_id");
+}
+
+TEST(type_prop, read_value_v6_init_shape_is_not_in_range) {
+    auto input = make_shared<ov::op::v0::Parameter>(element::f32, PartialShape{1, 2, 64, 64});
+
+    auto variable_info = op::util::VariableInfo{PartialShape{{2, 5}, {2, 5}, 64, 64}, element::f32, "variable_id"};
+    auto variable = std::make_shared<op::util::Variable>(variable_info);
+
+    std::shared_ptr<ov::op::v6::ReadValue> read_value;
+    EXPECT_ANY_THROW(read_value = std::make_shared<ov::op::v6::ReadValue>(input, variable));
+}
+
+TEST(type_prop, read_value_v6_init_shape_is_not_in_range_2) {
+    auto input = make_shared<ov::op::v0::Parameter>(element::f32, PartialShape{{1, 2}, 2, 64, 64});
+
+    auto variable_info = op::util::VariableInfo{PartialShape{1, 2, 64, 64}, element::f32, "variable_id"};
+    auto variable = std::make_shared<op::util::Variable>(variable_info);
+
+    std::shared_ptr<ov::op::v6::ReadValue> read_value;
+    EXPECT_ANY_THROW(read_value = std::make_shared<ov::op::v6::ReadValue>(input, variable));
```

Comment:
I don't see any tests for a newly created ctor with only variable id and without initialization node. Please cover this case.

Reply:
done